src/lib/components/BeforeAfter.react.js

```diff
@@ -1,55 +1,153 @@
+ * Before After Image Slider based on https://github.com/sneas/img-comparison-slider
  */
-export default class BeforeAfter extends Component {
+const BeforeAfter = props => {
+    const {
+        id,
+        before,
+        after,
+        width,
+        height,
+        hover,
+        value,
+        direction,
+        keyboard,
+        defaultProgress,
+        before_style,
+        after_style,
+        beforeClassName,
+        afterClassName,
+    } = props;
+
+    return (
+        <div id={id}>
+            <ImgComparisonSlider
+                hover={hover}
+                value={defaultProgress || value}
+                direction={direction}
+                keyboard={keyboard}
+            >
+                <img
+                    slot="first"
+                    width={width}
+                    height={height}
+                    src={before}
+                    style={before_style}
+                    className={beforeClassName}
+                />
+                <img
+                    slot="second"
+                    width={width}
+                    height={height}
+                    src={after}
+                    style={after_style}
+                    className={afterClassName}
+                />
+            </ImgComparisonSlider>
+        </div>
+    );
+};
+
+
+BeforeAfter.defaultProps = {
+    width: '100%',
+    height: 'auto',
+    hover: true,
+    value: 50,
+    direction: 'horizontal',
+    keyboard: 'enabled',
+};
 
-    render() {
-        return (
-            <BeforeAfterSlider {...this.props}/>
-        );
-    }
+BeforeAfter.propTypes = {
+    /**
+     * The ID used to identify this component in Dash callbacks.
+     */
+    id: PropTypes.string,
 
-}
+    /**
+     * Before image src
+     */
+    before: PropTypes.string.isRequired,
 
-BeforeAfter.propTypes = {
+    /**
+     * After image src
+     */
+    after: PropTypes.string.isRequired,
 
-    before: PropTypes.string,
+    /**
+     *  image height - default "auto" for responsive images.
+     */
+    height: PropTypes.string,
 
-    after: PropTypes.string,
+    /**
+     * image width - default "100%" for responsive images.
+     */
+    width: PropTypes.string,
 
-    width: PropTypes.number,
+    /**
+     * Automatic slide on mouse over.
+     */
+    hover: PropTypes.bool,
 
-    height: PropTypes.number,
+    /**
+     * The divider position can be specified as a percentage. 0 to 100
+     */
+    value: PropTypes.number,
 
-    defaultProgress: PropTypes.number,
+    /**
+     * Set slider direction
+     */
+    direction: PropTypes.oneOf(['horizontal', 'vertical']),
 
-    beforeClassName: PropTypes.string,
+    /**
+     * Enable/disable slider position control with the keyboard
+     */
+    keyboard: PropTypes.oneOf(['enabled', 'disabled']),
 
-    afterClassName: PropTypes.string,
+    /**
+     * Defines CSS styles of the before image.
+     */
+    before_style: PropTypes.object,
```

Comment:
Should we name the `before_style` prop as `beforeStyle` (and after vice versa) for consistency?

Reply:
I removed the `before_style` and `after_style` props  and went with your suggestion below of using `beforeProps` and `afterProps`  That's a good idea and makes all the `html.Img` props available without being super verbose.  

I think including an example in the docs of how to use these props will make it beginner friendly.